src/adaptation/AccessCheck.cc

```diff
@@ -129,11 +128,11 @@ Adaptation::AccessCheck::checkCandidates()
     while (!candidates.empty()) {
         if (AccessRule *r = FindRule(topCandidate())) {
             /* BUG 2526: what to do when r->acl is empty?? */
-            acl_checklist = new ACLFilledChecklist(r->acl, filter.request);
+            auto acl_checklist = ACLFilledChecklist::Make(r->acl, filter.request);
             acl_checklist->updateAle(filter.al);
             acl_checklist->updateReply(filter.reply);
             acl_checklist->syncAle(filter.request, nullptr);
-            acl_checklist->nonBlockingCheck(AccessCheckCallbackWrapper, this);
+            ACLFilledChecklist::NonBlockingCheck(std::move(acl_checklist), AccessCheckCallbackWrapper, this);
```

Comment:
I checked that neither gcc, no clang warn if a local variable (such as this pointer) is used after it has been moved. It looks like that [clang-tidy](https://stackoverflow.com/questions/72532377/g-detect-use-after-stdmove) may detect such situations (`bugprone-use-after-move` option), but I am not sure whether our build workflows utilize this tool.

Reply:
Our CI does not use clang-tidy today, but I have created an internal issue (SQUID-1000) to change that.